### ISSUE TITLE
Allow deterministic control of random outcomes

### ISSUE THREAD

**Issue by smallnamespace** [jleclanche/fireplace#323]
There are a few use cases for this:

1. Semi-deterministic tests, where we iterate through every possible random outcome (as opposed to manually patching up the outcome, or randomly generating many outcomes, where we might miss a case)
2. AI use cases, where the AI would like to explore random nodes in a particular direction (e.g. what's the best card I can draw? What's the best juggle/worst juggle that I can get?)
3. Syncing with HSReplays -- import of HSReplay into the engine and evolve it through time

**Comment by synap5e** [comment]
It may also be nice to have a probability assigned to each result node.

**Comment by smallnamespace** [comment]
Yeah that's a nice to have, might be tricky to do it in general though. I'm open to suggestions on how to go about implementing this, otherwise I'll take a good look through the code and draft something.

**Comment by smallnamespace** [comment]
Per discussion with jleclanche, would be good to think of this in the context of generalizing random outcomes into the DSL
